File: server/FitMate.Core/JsonModels/AdminAI/AIConversationQueryRequest.cs
using System.ComponentModel.DataAnnotations;
using FitMate.Core.JsonModels.Common;
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.AdminAI;

public class AIConversationQueryRequest : PagedRequest
{
    [StringLength(200)]
    public string? Search { get; set; }

    public long? UserId { get; set; }
    public AIConversationStatus? Status { get; set; }
    public DateTime? From { get; set; }
    public DateTime? To { get; set; }
}

File: client/src/types/JsonModels/AdminAI/AIConversationQueryRequest.ts
import type { JsonModels } from "../../backend";
import type { AIConversationStatus } from "../Enums/AIConversationStatus";

export interface AIConversationQueryRequest extends JsonModels.Common.PagedRequest
{
	search?: string;
	userId?: number;
	status?: AIConversationStatus;
	from?: string;
	to?: string;
}
